parquet/src/main/java/org/apache/iceberg/parquet/ParquetFilters.java

```diff
@@ -233,6 +266,29 @@ private static <C extends Comparable<C>> C getParquetPrimitive(Literal<?> lit) {
         "Type not supported yet: " + value.getClass().getName());
   }
 
+  @SuppressWarnings("unchecked")
+  private static <C extends Comparable<C>> Set<C> getParquetPrimitiveSet(Set<?> litSet) {
+    if (litSet == null) {
+      return Collections.emptySet();
+    }
+
+    // TODO: this needs to convert to handle BigDecimal and UUID
+    Set<C> convertedSet = Sets.newHashSet();
```

Comment:
if we add everything from litSet  to convertedSet, should we initialize it with the size parameter as Sets.newHashSet(litSet); It would be more efficient.